### PR TITLE
Update metadata indexing - Closes #1221

### LINKED ISSUE #1221: Update metadata indexing
### Description

Update the index schema and metadata indexing process as per the latest `app.json` and `nativetokens.json` schemas.

### Acceptance Criteria

- The `applications` and `tokens` index schemas are updated according to the above schema files
- The indexing logic is adjusted accordingly
- Tests are updated accordingly, if applicable
- The synchronization job runs every 10mins

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #1221 

### How was it solved?

- [x] The applications and tokens index schemas are updated according to the specs
- [x] Fix fee estimator
- [x] The indexing logic is adjusted accordingly
- [x] Update swagger specs
- [x] Indexer code cleanup
- [x] Update job `update.application.metadata` to perform updates every 10 mins

### How was it tested?
Local

### CODE REVIEW

File: services/blockchain-app-registry/shared/utils/downloadRepository.js [hunk 1]
```diff
@@ -151,7 +151,8 @@ const syncWithRemoteRepo = async () => {
 					await downloadFile(result.data.download_url, filePath);
 					logger.debug(`Successfully downloaded: ${file}`);
 
-					await indexMetadataFromFile(filePath);
+					const [, , network, appName, filename] = file.split('/');
+					await indexMetadataFromFile(network, appName, filename);
 					logger.debug('Successfully updated the database with the latest changes');
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/utils/downloadRepository.js:156]: Please execute the indexing logic only when the file has a `.json` extension.
```suggestion
					if (file.endsWith('.json')) {
						const [, , network, appName, filename] = file.split('/');
						await indexMetadataFromFile(network, appName, filename);
						logger.debug('Successfully updated the database with the latest changes');
					}
```

File: services/blockchain-app-registry/shared/metadataIndex.js [hunk 2]
```diff
@@ -50,25 +49,27 @@ const getTokensIndex = () => getTableInstance(
...
 const logger = Logger();
 
-const indexTokensInfo = async (filePath, dbTrx) => {
-	const tokensDB = await getTokensIndex();
+const [, , , , repo] = config.gitHub.appRegistryRepo.split('/');
+
+const FILENAME = {
+	APP_JSON: 'app.json',
+	NATIVETOKENS_JSON: 'nativetokens.json',
+};
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadataIndex.js:57]:
```suggestion
const FILENAME = Object.freeze({
	APP_JSON: 'app.json',
	NATIVETOKENS_JSON: 'nativetokens.json',
});
```

File: services/blockchain-app-registry/shared/metadataIndex.js [hunk 3]
```diff
@@ -50,25 +49,27 @@ const getTokensIndex = () => getTableInstance(
...
 
-	const response = await read(filePath);
-	const tokenInfo = JSON.parse(response);
+const KNOWN_CONFIG_FILES = Object.values(FILENAME);
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadataIndex.js:59]:
```suggestion
const KNOWN_CONFIG_FILES = Object.freeze(Object.values(FILENAME));
```

File: services/blockchain-app-registry/shared/metadataIndex.js [hunk 4]
```diff
@@ -50,25 +49,27 @@ const getTokensIndex = () => getTableInstance(
...
+
+const indexTokensInfo = async (tokenInfo, dbTrx) => {
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadataIndex.js:61]: Let us please rename `indexTokensInfo` to `indexTokensMetadata`. Similarly, do it for the chain/applications as well.

File: services/blockchain-app-registry/shared/metadataIndex.js [hunk 5]
```diff
@@ -78,77 +79,95 @@ const indexTokensInfo = async (filePath, dbTrx) => {
 	await tokensDB.upsert(tokenInfoToIndex, dbTrx);
 };
 
-const indexChainInfo = async (filePath, dbTrx) => {
+const indexChainInfo = async (chainInfo, dbTrx) => {
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadataIndex.js:82]:
```suggestion
const indexChainMetadata = async (chainInfo, dbTrx) => {
```

File: services/blockchain-app-registry/shared/metadataIndex.js [hunk 6]
```diff
@@ -78,77 +79,95 @@ const indexTokensInfo = async (filePath, dbTrx) => {
...
+		};
+
+		logger.debug(`Indexing tokens information for the app: ${app} (${network})`);
+		await indexTokensInfo(tokenInfo, dbTrx);
+		logger.debug(`Indexed tokens information for the app: ${app} (${network})`);
+	}
+	logger.info(`Finished indexing metadata information for the app: ${app} (${network})`);
 };
 
 const indexBlockchainMetadata = async () => {
```
Comment (sameersubudhi) [inline services/blockchain-app-registry/shared/metadataIndex.js:129]:
```suggestion
const indexAllBlockchainAppsMeta = async () => {
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/blockchain-app-registry/shared/database/schema/applications.js [hunk 7]
```diff
@@ -15,24 +15,17 @@
  */
 module.exports = {
 	tableName: 'applications',
-	primaryKey: ['name', 'network'],
+	primaryKey: ['network', 'chainName'],
```
Comment (priojeetpriyom) [inline services/blockchain-app-registry/shared/database/schema/applications.js:18]: why are we changing the primary key order?

**Review by priojeetpriyom — APPROVED** [review]

File: services/blockchain-app-registry/shared/metadataIndex.js [hunk 8]
```diff
@@ -50,109 +49,129 @@ const getTokensIndex = () => getTableInstance(
 
 const logger = Logger();
 
-const indexTokensInfo = async (filePath, dbTrx) => {
-	const tokensDB = await getTokensIndex();
+const [, , , , repo] = config.gitHub.appRegistryRepo.split('/');
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/blockchain-app-registry/shared/metadataIndex.js]: Done!

**Pull request merged**